ykrt/src/mt.rs

```diff
@@ -357,10 +252,19 @@ impl MT {
             let hl = unsafe { ls.hot_location() };
             let hl_ptr = hl as *mut _ as *mut ();
             match hl {
-                HotLocation::Compiled(tr) => {
+                HotLocation::Compiled(compiled_trace) => {
                     // FIXME: If we want to free compiled traces, we'll need to refcount (or use
                     // a GC) to know if anyone's executing that trace at the moment.
-                    todo!(); // FIXME run executable trace code here.
+
+                    // FIXME: this loop shouldn't exist. Trace stitching should be implemented in
+                    // the trace itself.
+                    // https://github.com/ykjit/yk/issues/442
+                    loop {
+                        eprintln!("jit-state: enter-jit-code");
+                        compiled_trace.exec(ctrlp_vars);
+                        eprintln!("jit-state: exit-jit-code");
+                    }
+                    // LAURIE: should the lock be held while the trace is being executed?
```

Comment:
No: in fact, the lock *mustn't* be held, otherwise the system will fail to make progress in some situations. For the time being it's fine to put the `unlock` before the `loop`.

Reply:
Good. That's what I thought.

Reply:
Fixed in 0df447df6